### PR TITLE
Fix build on macOS

### PR DESCRIPTION
**Pull request by endrift**
```sh
export PATH=$(brew --prefix llvm)/bin:$PATH
make
```

Fixes some random issues (commas where they shouldn't be, a handful of missing dependency links, etc), but the build system has innumerable issues and at this point and I'm done tearing out my hair.

### CODE REVIEW

**Comment by dvdfreitag** [comment]
A cursory glance of these changes look fine, but for some reason the Mephisto hello world test is hanging. I restarted the travis build, but it's hanging again on the second build.

Please make sure you're running these tests locally to verify your changes.

**Comment by endrift** [comment]
the Mephisto stuff is hanging due to a change to mephisto itself that @misson20000 made a few days ago (and this isn't the only PR whose build is failing as a result): https://github.com/reswitched/Mephisto/commit/210847a2bcadf6612c115a61c2aa69767c7a9847

cf. https://travis-ci.org/reswitched/libtransistor/builds/330093487 has the same failure

File: libtransistor.mk [hunk 1]
```diff
@@ -23,11 +26,11 @@ PYTHON3 := python3
 MEPHISTO := ctu
 RUBY := ruby
 COMPILER_RT_BUILTINS_LIB := $(LIBTRANSISTOR_HOME)/build/compiler-rt/lib/linux/libclang_rt.builtins-aarch64.a
-LIBTRANSISTOR_COMMON_LIBS := $(LIBTRANSISTOR_HOME)/build/newlib/aarch64-none-switch/newlib/libc.a $(COMPILER_RT_BUILTINS_LIB) $(LIBTRANSISTOR_HOME)/build/sdl2_install/lib/libSDL2.a $(LIBTRANSISTOR_HOME)/pthread/libpthread.a
+LIBTRANSISTOR_COMMON_LIBS := $(LIBTRANSISTOR_HOME)/build/newlib/aarch64-none-switch/newlib/libc.a $(LIBTRANSISTOR_HOME)/build/newlib/aarch64-none-switch/newlib/libm.a $(COMPILER_RT_BUILTINS_LIB) $(LIBTRANSISTOR_HOME)/pthread/libpthread.a
 LIBTRANSISTOR_NRO_LIB := $(LIBTRANSISTOR_HOME)/build/lib/libtransistor.nro.a
 LIBTRANSISTOR_NSO_LIB := $(LIBTRANSISTOR_HOME)/build/lib/libtransistor.nso.a
-LIBTRANSISTOR_NRO_LDFLAGS := --whole-archive $(LIBTRANSISTOR_NRO_LIB) --no-whole-archive $(LIBTRANSISTOR_COMMON_LIBS)
-LIBTRANSISTOR_NSO_LDFLAGS := --whole-archive $(LIBTRANSISTOR_NSO_LIB) --no-whole-archive $(LIBTRANSISTOR_COMMON_LIBS)
+LIBTRANSISTOR_NRO_LDFLAGS := --whole-archive $(LIBTRANSISTOR_NRO_LIB) --no-whole-archive $(LIBTRANSISTOR_COMMON_LIBS) -lSDL2
```
Comment (dvdfreitag) [inline libtransistor.mk]: Why did you add -lSDL2 here when sdl2 is already present in `LIBTRANSISTOR_COMMON_LIBS`?

https://github.com/reswitched/libtransistor/blob/8af3e728aa0f8b0d68809871f3d29cd384eb82ae/libtransistor.mk#L26

**Comment by endrift** [comment]
I'd taken it out of that list for a bit but I guess I put it back and forgot to remove -lSDL2

That said, building SDL2 shouldn't even be in this repo in the first place since it has nothing to do with the toolchain.

File: libtransistor.mk (hunk 1 above)
Comment (endrift) [reply libtransistor.mk]: Sorry, was on my phone and didn't notice this was a review. Reply below. (Thanks GitHub Mobile for being terrible)

**Comment by vgmoose** [comment]
Works for me! Thanks!
